addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -521,6 +522,84 @@ public void loadRssiValues() throws IOException {
         }
     }
 
+    @Override
+    public void setInstallMode(boolean enable, int seconds) throws IOException {
+        HmDevice gwExtrasHm = devices.get(HmDevice.ADDRESS_GATEWAY_EXTRAS);
+        
+        if (gwExtrasHm != null) {
+            // since the homematic virtual device exist: try setting install mode via its dataPoints
+            HmDatapoint installModeDataPoint = null;
+            HmDatapoint installModeDurationDataPoint = null;
+
+            // collect virtual datapoints to be accessed
+            for (HmChannel hmChannel : gwExtrasHm.getChannels()) {
+                HmDatapointInfo installModeDurationDataPointInfo = new HmDatapointInfo(HmParamsetType.VALUES, hmChannel,
+                        HomematicConstants.VIRTUAL_DATAPOINT_NAME_INSTALL_MODE_DURATION);
+                if (enable && hmChannel.hasDatapoint(installModeDurationDataPointInfo)) {
+                    installModeDurationDataPoint = hmChannel.getDatapoint(installModeDurationDataPointInfo);
+                }
+
+                HmDatapointInfo installModeDataPointInfo = new HmDatapointInfo(HmParamsetType.VALUES, hmChannel,
+                        HomematicConstants.VIRTUAL_DATAPOINT_NAME_INSTALL_MODE);
+
+                if (hmChannel.hasDatapoint(installModeDataPointInfo)) {
+                    installModeDataPoint = hmChannel.getDatapoint(installModeDataPointInfo);
+                }
+            }
+            for (VirtualDatapointHandler dpHandler : virtualDatapointHandlers) {
```

Comment:
What is the loop for?

Reply:
Thanks for noticing this. This loop is totally useless, since the handler is picked in the `getVirtualDatapointHandler(..)` method anyways. Actually I introduced an unused variable here, but since there is an `instanceof` check, I didn't get warning....